src/Firebase/IdentityPlatform.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kreait\Firebase;
+
+use Kreait\Firebase\IdentityPlatform\ApiClient;
+use Kreait\Firebase\IdentityPlatform\DefaultSupportedIdpConfig;
+use Kreait\Firebase\IdentityPlatform\InboundSamlConfig;
+use Kreait\Firebase\IdentityPlatform\OAuthIdpConfig;
+use Kreait\Firebase\Request\DefaultSupportedIdpConfig as DefaultSupportedIdpConfigRequest;
+use Kreait\Firebase\Request\InboundSamlConfig as InboundSamlConfigRequest;
+use Kreait\Firebase\Request\OAuthIdpConfig as OAuthIdpConfigRequest;
+use Kreait\Firebase\Util\ArrayHelper;
+use Kreait\Firebase\Util\JSON;
+use Psr\Http\Message\ResponseInterface;
+
+class IdentityPlatform implements Contract\IdentityPlatform
+{
+    private ApiClient $client;
+
+    /**
+     * @param iterable<ApiClient|null>|ApiClient|null ...$x
+     *
+     * @internal
+     */
+    public function __construct(...$x)
```

Comment:
I know I did this in the `Auth` class, but this was only because it was one of the oldest classes and I needed it to be backward compatible at the time - you can inject the dependencies directly here without using the `foreach` construct 🙏